openvdb/openvdb/python/CMakeLists.txt

```diff
@@ -74,11 +74,7 @@ endfunction()
 #   target but this was only added in CMake 3.15. See:
 #      https://github.com/AcademySoftwareFoundation/openvdb/issues/886
 set(OPENVDB_PYTHON_DEPS)
-set(OPENVDB_PYTHON_REQUIRED_COMPONENTS Development)
-
-if(OPENVDB_BUILD_PYTHON_UNITTESTS)
-  list(APPEND OPENVDB_PYTHON_REQUIRED_COMPONENTS Interpreter)
-endif()
+set(OPENVDB_PYTHON_REQUIRED_COMPONENTS Development Interpreter)
```

Comment:
Was there a reason you changed this? Technically, the `Interpreter` should not be needed unless the unit tests are being run

Reply:
I totally agree, but the component name "Interpreter" is somehow misleading. The variable `${Python_SITELIB}` will only get populated if we include the `Intepreter` component. Is not about accessing the python interpreter per see. 

That variable gets created at https://github.com/Kitware/CMake/blob/859241d2bbaae83f06c44bc21ab0dbd38725a3fc/Modules/FindPython/Support.cmake#L1830

Reply:
Hmm we previously always searched for both but we changed it due to this issue:

https://github.com/AcademySoftwareFoundation/openvdb/issues/621

@e4lam do you remember what the exact side effects were? The issue refers to the fact that the `Interpreter` target forces another dependency, but I'm unclear how problematic this actually is. For context, this PR aims to improve the default installation of the python module by using the `Python_SITELIB` CMake variable... which is only set if you look for the `Interpreter` component

Reply:
Sorry, I don't really remember the exact error. AFAIR, building the openvdb vcpkg would fail until I removed `Interpreter` there. I don't recall now if vcpkg simply never built the python interpreter at the time. So when I tried building openvdb, cmake would fail because it couldn't find a working python interpreter since I didn't build anything in vcpkg that required the python interpreter. Looking this up now, I see https://github.com/microsoft/vcpkg/pull/14891 has since occurred, so maybe if we rejigged the openvdb vcpkg dependencies, it might be possible to add this back in again. Not sure if I can find time to try this though.

Reply:
Thanks a lot @e4lam! It's weird that we need the interpreter for this, but I guess it's running some python to figure out the correct installation path. If that's the best and most correct python-y way of getting it then seems like we'll need it - if the fix in vcpkg only went in for py3 then we can re-jig this so that it's only brought in for in py3 and later. 

Additionally, if a user has provided their own absolute file path for the vdb cmake variable `PYOPENVDB_INSTALL_DIRECTORY` (i.e. `-DPYOPENVDB_INSTALL_DIRECTORY=/my/custom/path`) then we can also avoid searching for `Interpreter`. That logic should help mitigate issues. @nachovizzo does that make sense to you too?

Reply:
It's weird indeed and I don't like it either. But looks like that's how it's done on the python world :P. To dive even deeper you can take a look at [this line@cmake v.3.12](https://github.com/Kitware/CMake/blob/f84c15ef2fa30dd074fcccafccec6b9b69266619/Modules/FindPython/Support.cmake#L329). Actually the one pointed out above it was the wrong line to look for.

If we decrypt a bit what they are running in python looks like:
```python
import sys
from distutils import sysconfig

sys.stdout.write(
    ";".join(
        [
            sysconfig.get_python_lib(plat_specific=False, standard_lib=True),
            sysconfig.get_python_lib(plat_specific=True, standard_lib=True),
            sysconfig.get_python_lib(plat_specific=False, standard_lib=False),
            sysconfig.get_python_lib(plat_specific=True, standard_lib=False),
        ]
    )
)
```

Which in my `Ubuntu 20.04` installation outputs:
```sh
/usr/lib/python3.8;/usr/lib/python3.8;/usr/lib/python3/dist-packages;/usr/lib/python3/dist-packages
```

But when using the docker image for the CI builds this is the output
```sh
$ docker run --rm -it aswf/ci-openvdb python -c '
import sys
from distutils import sysconfig

sys.stdout.write(
    ";".join(
        [
            sysconfig.get_python_lib(plat_specific=False, standard_lib=True),
            sysconfig.get_python_lib(plat_specific=True, standard_lib=True),
            sysconfig.get_python_lib(plat_specific=False, standard_lib=False),
            sysconfig.get_python_lib(plat_specific=True, standard_lib=False),
        ]
    )
)'
/usr/local/lib/python2.7;/usr/local/lib/python2.7;/usr/local/lib/python2.7/site-packages;/usr/local/lib/python2.7/site-packages
```

As you can see they are different locations, and **apparently** the only way of getting to know this is by launching a python shell and getting the `sysconfig` variables.

Aout the `PYOPENVDB_INSTALL_DIRECTORY`, I agree and fixed it in de283e4eada4465